test/installation_node/install hook

```diff
@@ -19,6 +19,13 @@ fail() {
 
 ! nvm_is_version_installed "${VERSION}" || nvm uninstall "${VERSION}" || die 'uninstall failed'
 
+# an existing but empty VERSION_PATH directory should not be enough to satisfy nvm_is_version_installed
+mkdir -p "${VERSION_PATH}"
+if [ -z "$(ls -A ${VERSION_PATH})" ]; then
```

Comment:
when would the dir be non-empty here, and wouldn't we want to fail the test if this is the case?

Reply:
Are you saying that the `-z` test on `ls -A` for empty dir is unnecessary? I guess in other words, am I right to assume that the `$VERSION` is uninstalled after line 20 finishes?

Reply:
If uninstallation fails, line 20 would exit.

You could `rm -rf "${VERSION_PATH} && mkdir -p "${VERSION_PATH}"` instead, though, just in case?

Reply:
Right, so assuming uninstallation is successful, `$VERSION_PATH` shouldn't even exist after line 20. I did initially consider a `rm -rf` but opted for a sanity check instead - that's what the if was for.

If we're 100% confident about the correctness of the non-zero exit code from `nvm uninstall`, then we don't need to do anything just in case, and can remove the if statement altogether. What do you think?

Reply:
With the if statement, with a non-empty folder, it silently skips this test. That's not good.
With nothing, with a non-empty folder, it silently tests the wrong thing. That's not good either.

In other words, if the precondition fails, either the whole test should blow up, or, it should be silently cleaned up so that the next test is actually testing something.

Reply:
Good points! I can't really think of a better approach at the moment. 5b8da24